File: source/filemanager/plugin_src/ai/rorsch/pandagenie/nativelib/FileManagerLib.java
package ai.rorsch.pandagenie.nativelib;

public class FileManagerLib {

    static {
        System.loadLibrary("filemanager");
    }

    public native String nativeListDirectory(String path);
    public native boolean nativeCreateDirectory(String path);
    public native boolean nativeDeleteFile(String path);
    public native boolean nativeDeleteDirectory(String path, boolean recursive);
    public native boolean nativeCopyFile(String src, String dst);
    public native boolean nativeMoveFile(String src, String dst);
    public native boolean nativeRenameFile(String oldPath, String newPath);
    public native String nativeGetFileInfo(String path);
    public native String nativeSearchFiles(String dir, String pattern, boolean recursive);
    public native String nativeReadTextFile(String path);
    public native boolean nativeWriteTextFile(String path, String content);
    public native boolean nativeFileExists(String path);
    public native long nativeGetFileSize(String path);
}

File: source/filemanager/jni_bridge/ai/rorsch/pandagenie/nativelib/FileManagerLib.kt
package ai.rorsch.pandagenie.nativelib

class FileManagerLib {

    external fun nativeListDirectory(path: String): String
    external fun nativeCreateDirectory(path: String): Boolean
    external fun nativeDeleteFile(path: String): Boolean
    external fun nativeDeleteDirectory(path: String, recursive: Boolean): Boolean
    external fun nativeCopyFile(src: String, dst: String): Boolean
    external fun nativeMoveFile(src: String, dst: String): Boolean
    external fun nativeRenameFile(oldPath: String, newPath: String): Boolean
    external fun nativeGetFileInfo(path: String): String
    external fun nativeSearchFiles(dir: String, pattern: String, recursive: Boolean): String
    external fun nativeReadTextFile(path: String): String
    external fun nativeWriteTextFile(path: String, content: String): Boolean
    external fun nativeFileExists(path: String): Boolean
    external fun nativeGetFileSize(path: String): Long
}
